p2p_wallet/Services/Repositories/BaseRepository/ListRepository/ListRepository.swift

```diff
@@ -27,63 +26,39 @@ class ListRepository<ItemType: Hashable>: ItemRepository<[ItemType]> {
     // MARK: - Actions
 
     /// Erase data and reset repository to its initial state
-    override func flush() {
+    override func reset(autoFetch: Bool = true) {
         paginationStrategy?.resetPagination()
-        super.flush()
-    }
-    
-    /// Erase and reload all data
-    override func reload() {
-        super.reload()
+        super.reset(autoFetch: autoFetch)
     }
     
-    /// Refresh data
-    override func refresh() {
-        // if pagination enabled, erase and reload all data
-        if let paginationStrategy {
-            paginationStrategy.resetPagination()
-            super.reload()
-        }
-        
-        // if pagination is not enabled, just refresh data without erasing current data
-        else {
-            super.refresh()
-        }
-    }
-    
-    /// Indicate if should fetch new data to prevent unwanted request
+    /// Indicate if repository can fetch more data
     /// - Returns: should fetch new data
-    override func shouldRequest() -> Bool {
-        var shouldRequest = super.shouldRequest()
-        
-        // check if isLastPageLoaded
-        if let paginationStrategy {
-            shouldRequest = shouldRequest && !paginationStrategy.isLastPageLoaded
+    private func fetchable() -> Bool {
```

Comment:
Let's move private data to the bottom of the class, in section like `MARK: - Helpers`